src/npcmove.cpp

```diff
@@ -1704,6 +1712,28 @@ void npc::move_pause()
     }
 }
 
+static tripoint nearest_passable( const tripoint &p, const tripoint &closest_to )
```

Comment:
This would look good inside the `map` class. Besides, it would be great to have some unification with `map::adjacent_tile()`. It does similar work, but it's richer since it checks traps, dangerous fields, and other npcs.